drivers/gles3/storage/texture_storage.cpp

```diff
@@ -1779,7 +1779,63 @@ void TextureStorage::_create_render_target_backbuffer(RenderTarget *rt) {
 		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
 	}
 }
+void GLES3::TextureStorage::copy_scene_to_backbuffer(RenderTarget *rt, const bool uses_screen_texture, const bool uses_depth_texture) {
+	if (rt->backbuffer_fbo != 0) {
+		return;
+	}
+
+	Config *config = Config::get_singleton();
+	bool use_multiview = rt->view_count > 1 && config->multiview_supported;
+	GLenum texture_target = use_multiview ? GL_TEXTURE_2D_ARRAY : GL_TEXTURE_2D;
+
+	glGenFramebuffers(1, &rt->backbuffer_fbo);
+	glBindFramebuffer(GL_FRAMEBUFFER, rt->backbuffer_fbo);
+	if (rt->backbuffer == 0 && uses_screen_texture) {
+		glGenTextures(1, &rt->backbuffer);
+		glBindTexture(texture_target, rt->backbuffer);
+		if (use_multiview) {
+			glTexImage3D(texture_target, 0, rt->color_internal_format, rt->size.x, rt->size.y, rt->view_count, 0, rt->color_format, rt->color_type, nullptr);
```

Comment:
If using multiview, I don't think the backbuffer can be shared with 2D as in 2D the backbuffer is always assumed to be a Texture_2D. We may need to create a separate backbuffer for multiview

@dsnopek How does 2D rendering work in XR when using Multiview? Does it just render to the first view index, or does it require using both views? From my brief look at the shader code and canvas render code it looks like 2D rendering should still just take place on a texture2D

Reply:
> How does 2D rendering work in XR when using Multiview?

That's a good question! We probably want to just disable 2D rendering on multiview XR viewports, but I'll need to test to see what is actually happening now.

(We've got to be careful to NOT disable 2D rendering for non-multiview XR apps, since smartphone AR still needs to be able to render 2D stuff.)

The problem is that (a) rendering to just the first view, or (b) rendering the exact same thing to all views, are both wrong. :-) We'd need to render something slightly different to each view to get the right stereoscopic effect.

For the most part, stereo XR apps usually render 2D stuff onto a mesh in the 3D world, rather than rendering directly on to the output FBO.

Reply:
I just tried putting a CanvasLayer with a ColorRect on it into a VR test app with gl_compatibility, and it didn't show up at all!

Reply:
is there a way I could test without needing a VR headset?

Reply:
> is there a way I could test without needing a VR headset?

On Linux, you can use Monado to have a simulated OpenXR headset, which you control with keyboard and mouse. I'm not sure what's possible on other platforms.

Or, you can also use the `MobileVRInterface`, which just renders the two views to the screen, without needing a headset.

BTW, my last comment about the ColorRect not showing was a good thing! We don't want 2D stuff to render on top of 3D with stereoscopic XR.

Reply:
For the purposes of this PR I guess we can assume that if ``use_multiview`` is true, we can just forget about 2D rendering. 

Therefore, in this section when ``use_multiview`` is false we can call ``_create_render_target_backbuffer`` to create the backbuffer with full mipmaps, so it can be shared with 2D